### PR TITLE
Expand Compose UI test coverage for market list, detail, and app navigation

### PR DESCRIPTION
**Pull request by masoudkarimi**
## Summary
- Add Compose UI tests for favorite market list states, item actions, swipe-to-remove, pull refresh, and multi-item rendering.
- Add paged market list tests for loaded data, refresh errors, append loading/error states, item clicks, and favorite clicks.
- Add market detail screen tests for favorite actions, stat rows, loading state behavior, and empty chart data.
- Add app-level instrumentation test setup with a custom runner, test application, Koin test module, and fake market repository.
- Add a MainActivity instrumentation test that navigates to the Favorite screen and verifies the empty favorite state.
- Add a test tag for the paging append loading indicator.
- Handle empty chart data in `QuadLineChart` to avoid division-by-zero and show a clear empty state.
- Correct favorite item swipe direction for dismiss behavior.

## Verification
- `./gradlew :app:connectedDebugAndroidTest -Pandroid.testInstrumentationRunnerArguments.class=ir.composenews.ui.MainActivityTest`

## Notes
- `gradle/gradle-daemon-jvm.properties` is untracked local Gradle state and should not be included in the PR.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -22,7 +22,7 @@ android {
         versionCode = libs.versions.projectVersionCode.get().toInt()
         versionName = libs.versions.projectVersionName.get()
 
-        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner = "ir.composenews.ui.InstrumentationTestRunner"
```
Comment (Kaaveh) [inline app/build.gradle.kts:25]: Why do we need this class? What was the AndroidJUnitRunner problem?

File: core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package ir.composenews.core_test.dispatcher;
+
+import ir.composenews.base.dispatcher.DispatcherProvider;
+import kotlinx.coroutines.CoroutineDispatcher;
+import kotlinx.coroutines.test.TestCoroutineDispatchersKt;
+import kotlinx.coroutines.test.TestCoroutineScheduler;
+import kotlinx.coroutines.test.TestDispatcher;
+
+public final class TestDispatcherProvider implements DispatcherProvider {
```
Comment (Kaaveh) [inline core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.java:9]: We can't migrate it to Kotlin, right?

File: data/market-local/src/test/java/ir/composenews/localdatasource/database/MarketDaoTest.kt [hunk 3]
```diff
@@ -6,14 +6,14 @@
...
+import io.kotest.core.spec.style.FreeSpec
 import io.kotest.matchers.collections.shouldContainExactly
 import io.kotest.matchers.shouldBe
-import ir.composenews.core_test.SuspendSpec
 import ir.composenews.db.MarketDatabase
 import ir.composenews.db.MarketEntity
 import kotlinx.coroutines.flow.first
 
-class MarketDaoTest : SuspendSpec({
+class MarketDaoTest : FreeSpec({
```
Comment (Kaaveh) [inline data/market-local/src/test/java/ir/composenews/localdatasource/database/MarketDaoTest.kt:16]: Why did the scope of this class change? For what reason?

File: app/src/androidTest/java/ir/composenews/ui/TestApplication.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package ir.composenews.ui
+
+import android.app.Application
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+
+class TestApplication : Application() {
```
Comment (Kaaveh) [inline app/src/androidTest/java/ir/composenews/ui/TestApplication.kt:7]: Were this test application registered?

File: app/src/androidTest/java/ir/composenews/ui/TestModule.kt [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        factory { GetMarketByIdUseCase(get()) }
+        factory { GetMarketChartUseCase(get()) }
+        factory { GetMarketDetailUseCase(get()) }
+        single<DispatcherProvider> {
+            object : DispatcherProvider {
+                override val ui: CoroutineDispatcher = Dispatchers.Unconfined
+                override val io: CoroutineDispatcher = Dispatchers.Unconfined
+                override val bg: CoroutineDispatcher = Dispatchers.Unconfined
+            }
+        }
```
Comment (Kaaveh) [inline app/src/androidTest/java/ir/composenews/ui/TestModule.kt:30]: I guess you have been defining a Java version of this dispatcher for a while. Can we reuse them?

File: feature/marketdetail/src/androidTest/java/ir/composenews/marketdetail/MarketDetailScreenTest.kt [hunk 6]
```diff
@@ -32,21 +34,25 @@ class MarketDetailScreenTest {
...
+                return flowOf(null)
             }
 
+            override suspend fun syncMarketList() = Unit
+
+            override suspend fun toggleFavoriteMarket(oldMarket: Market) = Unit
+
             override fun fetchChart(id: String): Flow<Resource<MarketChart, Errors>> {
                 TODO("Not yet implemented")
             }
```
Comment (Kaaveh) [inline feature/marketdetail/src/androidTest/java/ir/composenews/marketdetail/MarketDetailScreenTest.kt]: We have another use case for this fake repository. I think it's better to move it to its dedicated module where it's defined.

File: library/designsystem/src/main/java/ir/composenews/designsystem/component/QuadLineChart.kt [hunk 7]
```diff
@@ -73,6 +75,17 @@ fun QuadLineChart(
...
+        if (data.isEmpty()) {
+            Text(
+                modifier = Modifier
+                    .align(Alignment.Center),
+                text = "No chart data",
+                style = MaterialTheme.typography.bodyMedium
+            )
+            return@Box
+        }
+
```
Comment (Kaaveh) [inline library/designsystem/src/main/java/ir/composenews/designsystem/component/QuadLineChart.kt:88]: That's wonderful, but for these kinds of changes that are not related to the actual task, please create another tiny PR.

File: app/build.gradle.kts (hunk 1 above)
Comment (masoudkarimi) [reply app/build.gradle.kts:25]: This test runner will register a TestApplication class that I added here 
`app/src/androidTest/java/ir/composenews/ui/TestApplication.kt
`
That application class will help us registering test modules when starting the Koin and also calling stopKoin when the application is terminated.

File: core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.java (hunk 2 above)
Comment (masoudkarimi) [reply core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.java:9]: As far as I know TestFixture doesn't support Kotlin codes.

File: app/src/androidTest/java/ir/composenews/ui/TestApplication.kt (hunk 4 above)
Comment (masoudkarimi) [reply app/src/androidTest/java/ir/composenews/ui/TestApplication.kt:7]: It's being used in `ir.composenews.ui.InstrumentationTestRunner`

File: data/market-local/src/test/java/ir/composenews/localdatasource/database/MarketDaoTest.kt (hunk 3 above)
Comment (masoudkarimi) [reply data/market-local/src/test/java/ir/composenews/localdatasource/database/MarketDaoTest.kt:16]: Good catch. 
This changed as part of the core/test migration to `testFixtures`. `SuspendSpec` was a custom wrapper from `core_test`, but it was removed during that migration, so this test now uses Kotest’s standard `FreeSpec` directly.
For `MarketDaoTest`, `SuspendSpec` was not providing any required behavior. The suspend calls are inside beforeEach / test bodies, and **FreeSpec already supports suspend test callbacks**. So this keeps the test behavior the same while removing the dependency on the custom deleted spec class.

File: app/src/androidTest/java/ir/composenews/ui/TestModule.kt (hunk 5 above)
Comment (masoudkarimi) [reply app/src/androidTest/java/ir/composenews/ui/TestModule.kt:30]: Good catch again :). I fixed it.

File: feature/marketdetail/src/androidTest/java/ir/composenews/marketdetail/MarketDetailScreenTest.kt (hunk 6 above)
Comment (masoudkarimi) [reply feature/marketdetail/src/androidTest/java/ir/composenews/marketdetail/MarketDetailScreenTest.kt]: It's a good idea
I moved the fake MarketRepository implementation into a shared FakeMarketRepository so both app-level and market detail UI tests can reuse the same test data source instead of duplicating local fake implementations.

File: library/designsystem/src/main/java/ir/composenews/designsystem/component/QuadLineChart.kt (hunk 7 above)
Comment (masoudkarimi) [reply library/designsystem/src/main/java/ir/composenews/designsystem/component/QuadLineChart.kt:88]: You are totally right. I also worte test for it. If I remove it I need to also remote the test. What do you think?

Comment (Kaaveh) [reply library/designsystem/src/main/java/ir/composenews/designsystem/component/QuadLineChart.kt:88]: No, we keep it right now. Thanks.

File: domain/market/src/main/java/ir/composenews/domain/repository/MarketRepository.kt [hunk 8]
```diff
@@ -22,3 +24,55 @@ interface MarketRepository {
     fun fetchChart(id: String): Flow<Resource<MarketChart, Errors>>
     fun fetchDetail(id: String): Flow<Resource<MarketDetail, Errors>>
 }
+
+class FakeMarketRepository : MarketRepository {
```
Comment (Kaaveh) [inline domain/market/src/main/java/ir/composenews/domain/repository/MarketRepository.kt]: Can't it be moved into the test folder for separation from production code?

Comment (masoudkarimi) [reply domain/market/src/main/java/ir/composenews/domain/repository/MarketRepository.kt]: I moved it to the test fixtures source set. Thanks for pointing that out.

**Review by Kaaveh — APPROVED** [review]
